src/components/AppProvider/AppProvider.tsx

```diff
@@ -65,25 +67,32 @@ export default class AppProvider extends React.Component<Props, State> {
     // eslint-disable-next-line react/no-did-update-set-state
     this.setState({
       context: createAppProviderContext({
-        i18n,
         linkComponent,
         apiKey,
         shopOrigin,
         forceRedirect,
         stickyManager: this.stickyManager,
       }),
+      intl: new Intl(i18n),
     });
   }
 
   render() {
     const {theme = {logo: null}, children} = this.props;
-    const {context} = this.state;
+    const {
+      context: {...appProviderContext},
+      intl,
+    } = this.state;
 
     return (
-      <AppProviderContext.Provider value={context}>
-        <ThemeProvider theme={theme}>
-          {React.Children.only(children)}
-        </ThemeProvider>
+      <AppProviderContext.Provider value={appProviderContext}>
+        <IntlContext.Provider value={intl}>
+          <ScrollLockManagerContext.Provider value={this.scrollLockManager}>
+            <ThemeProvider theme={theme}>
```

Comment:
Should we also be updating `ThemeProvider` to follow the same syntax? Something like `Theme.Provider`? Or do we want to preserve the current syntax to maintain backwards compatibility?

Reply:
This is an interesting one. In all the other cases here AppProviderContext/IntlContext/ScrollLockManagerContext are react contexts, while the ThemeProvider is a component of our creation, which acts pretty similar to a react context provider but we add that extra div to add the theme custom properties. My gut feel is that we don't want to make this mimic the API of a context because it's not a context.

Moving the ThemeProvider component into utilities/theme feels strange as we're putting a component into utilities but it probably serves our separation of concerns better than keeping the ThemeProvider as a "private" component in components.